src/Symfony/Component/AssetMapper/ImportMap/ImportMapVersionChecker.php

```diff
@@ -0,0 +1,131 @@
+            if (!$entry->isRemotePackage()) {
+                continue;
+            }
+
+            $dependencies = $this->packageDownloader->getDependencies($entry->importName);
+            if (!$dependencies) {
+                continue;
+            }
+
+            $packageName = $this->extractPackageNameFromImport($entry->importName);
+
+            $url = str_replace(
+                ['%package%', '%version%'],
+                [$packageName, $entry->version],
+                self::PACKAGE_METADATA_PATTERN
+            );
+            $packages[$packageName] = [
+                $this->httpClient->request('GET', $url),
+                $dependencies,
+            ];
+        }
+
+        $errors = [];
+        $problems = [];
+        foreach ($packages as $packageName => [$response, $dependencies]) {
+            if (200 !== $response->getStatusCode()) {
+                $errors[] = [$packageName, $response];
+                continue;
+            }
+
+            $data = json_decode($response->getContent(), true);
+            // dependencies seem to be found in both places
+            $packageDependencies = array_merge(
+                $data['dependencies'] ?? [],
+                $data['peerDependencies'] ?? []
+            );
+
+            foreach ($dependencies as $dependencyName) {
+                $dependencyVersionConstraint = $packageDependencies[$dependencyName] ?? null;
+
+                if (null === $dependencyVersionConstraint) {
+                    // odd that this would happen
+                    continue;
+                }
+
+                $dependencyPackageName = $this->extractPackageNameFromImport($dependencyName);
+                if (!$entries->has($dependencyPackageName)) {
+                    $problems[] = new PackageVersionProblem($packageName, $dependencyName, $dependencyVersionConstraint, null);
+
+                    continue;
+                }
+
+                if (!$this->isVersionSatisfied($dependencyVersionConstraint, $entries->get($dependencyName)->version)) {
+                    $problems[] = new PackageVersionProblem($packageName, $dependencyName, $dependencyVersionConstraint, $entries->get($dependencyName)->version);
+                }
+            }
+        }
+
+        try {
+            ($errors[0][1] ?? null)?->getHeaders();
+        } catch (HttpExceptionInterface $e) {
+            $response = $e->getResponse();
+            $packageNames = implode('", "', array_column($errors, 0));
+
+            throw new RuntimeException(sprintf('Error %d finding metadata for package "%s". Response: ', $response->getStatusCode(), $packageNames).$response->getContent(false), 0, $e);
+        }
+
+        return $problems;
+    }
+
+    private function isVersionSatisfied(string $versionConstraint, ?string $version): bool
+    {
+        if (null === $version) {
+            return false;
+        }
+
+        return Semver::satisfies($version, $versionConstraint);
```

Comment:
Could easily be a one-line:

```suggestion
        return $version ? Semver::satisfies($version, $versionConstraint) : false;
```